rebuttal.tex

```diff
@@ -0,0 +1,948 @@
+
+Table~\ref{P-features-table} was extended to include submodule names.
+\end{solution}
+
+\question The sets support listing is unnaturally split in two by the ``This includes\ldots'' sentence which would fit better in parentheses.
+\begin{solution}
+This suggestion was implemented.
+\end{solution}
+
+\question Line 355, add a sentence explaining that in SymPy \texttt{str == repr}, because in Python \texttt{repr} is used to get an unambiguous valid Python code representation, while the return value of \texttt{str} is meant to be human-readable.
+\begin{solution}
+See the added footnote~\ref{P-note:repr}.
+\end{solution}
+
+\question Lines 359 and 379: what is 2D text representation? It seems that ``2D'' shouldn't be here.
+\begin{solution}
+This sentence was replaced with:
+\begin{quote}
+A two-dimensional (2D) textual representation of the expression can
+be printed with monospace fonts via \verb|pprint|.
+\end{quote}
+We use here the same terminology as in the Mathematica
+online documentation, e.g., the
+\href{https://reference.wolfram.com/language/ref/OutputForm.html}{\texttt{OutputForm}}
+function:
+\begin{quote}
+\texttt{OutputForm[expr]}: prints as a two-dimensional representation of expr using only keyboard characters.
+\end{quote}
+\end{solution}
+
+\question Line 427: every dictionary is a ``dictionary of keys''. This should be a dictionary with coordinate tuples as keys associated with the appropriate values.
+\begin{solution}
+``Dictionary of Keys'' (DOK) refers specifically to a sparse matrix
+representation where entries are stored as \texttt{(row, column)} pairs
+mapping to the elements. See for instance \texttt{scipy.sparse.dok\_sparse}. We
+have fixed the capitalization of ``Dictionary of Keys'' and added the
+abbreviation (DOK), and stated explicitly what is meant by this. We have also
+made a similar note about List of Lists (LIL) earlier in the section.
+\end{solution}
+
+\question Section 4 would benefit from an introduction, and lines 440-- should become a new subsection 4.1. (named ``Float'' or ``Real numbers support'' or similar).
+\begin{solution}
+We have added an introduction and a subsection~\ref{P-sec:floating-point}, ``Floating-Point Numbers''
+(which, for mathematical clarity, we consider to be distinct from ``real
+numbers'').
+\end{solution}
+
+\question I suggest a better example for lines 459--460: a computation of
+$(e^{100} +1) - e^{100}$:
+
+\begin{verbatim}
+   >>> (exp(100)+1).evalf() - exp(100).evalf()
+   0
+   >>> ((exp(100)+1) - exp(100)).evalf()
+   1.00000000000000
+   >>> (exp(100)+1) - exp(100)
+   1
+\end{verbatim}
+
+or two different ways to compute the 100th Fibonacci number:
+
+\begin{verbatim}
+   >>> phi = (1+sqrt(5))/2
+   >>> psi = (1-sqrt(5))/2
+   >>> ((phi**100-psi**100)/sqrt(5)).evalf()-fibonacci(100)
+   65536.0000000000
+   >>> ((phi**100-psi**100)/sqrt(5) - fibonacci(100)).evalf()
+   0.e-104
+\end{verbatim}
```

Comment:
@fredrik-johansson how do you feel about the example currently in the paper (`cos(exp(-100)) - 1`) vs. this one? The paper states that this is a "notorious case". Is this expression a classic example, which can be cited from somewhere?